redirector/go/main.go

```diff
@@ -56,14 +56,35 @@ type cacheEntry struct {
 }
 
 type root struct {
+	mpregex *regexp.Regexp
+
 	lock            sync.RWMutex
 	mountpointCache map[uint32]cacheEntry
+
+	getMountsLock sync.Mutex
 }
 
-func newRoot() *root {
+func newRoot() (*root, error) {
+	runmodeStr := "keybase"
+	switch os.Getenv("KEYBASE_RUN_MODE") {
+	case "staging":
+		runmodeStr = "keybase.staging"
+	case "devel":
+		runmodeStr = "keybase.devel"
+	}
+
+	// Find mountpoints like "/home/user/.local/share/keybase/fs", or
+	// "/home/user/keybase", and make sure it doesn't match mounts for
+	// another run mode, say "/home/user/keybase.staging".
+	rx, err := regexp.Compile(fmt.Sprintf("/%s/|%s$", runmodeStr))
```

Comment:
This `Sprintf` takes 2 strings, but only passes in one.

Reply:
Additionally, the runmode string can have a `.` in it, but in regex land that means any character.